src/auth/src/credentials/external_account.rs

```diff
@@ -825,6 +831,42 @@ impl ProgrammaticBuilder {
         self
     }
 
+    /// Sets the optional service account impersonation URL.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use google_cloud_auth::credentials::external_account::ProgrammaticBuilder;
+    /// # use google_cloud_auth::credentials::subject_token::{SubjectTokenProvider, SubjectToken, Builder as SubjectTokenBuilder};
+    /// # use google_cloud_auth::errors::SubjectTokenProviderError;
+    /// # use std::error::Error;
+    /// # use std::fmt;
+    /// # use std::sync::Arc;
+    /// #
+    /// # #[derive(Debug)]
+    /// # struct MyTokenProvider;
+    /// #
+    /// # #[derive(Debug)]
+    /// # struct MyProviderError;
+    /// # impl fmt::Display for MyProviderError { fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { write!(f, "MyProviderError") } }
+    /// # impl Error for MyProviderError {}
+    /// # impl SubjectTokenProviderError for MyProviderError { fn is_transient(&self) -> bool { false } }
+    /// #
+    /// # impl SubjectTokenProvider for MyTokenProvider {
+    /// #     type Error = MyProviderError;
+    /// #     async fn subject_token(&self) -> Result<SubjectToken, Self::Error> {
+    /// #         Ok(SubjectTokenBuilder::new("my-programmatic-token".to_string()).build())
+    /// #     }
+    /// # }
+    /// # let provider = Arc::new(MyTokenProvider);
+    /// let builder = ProgrammaticBuilder::new(provider)
+    ///     .with_service_account_impersonation_url("my-service-account-impersonation-url");
+    /// ```
+    pub fn with_service_account_impersonation_url<S: Into<String>>(mut self, url: S) -> Self {
```

Comment:
We should get `target_principal` instead of the whole url. 
1. It is unnecessary for the user to figure out the whole url
2. when we introduce universe domain, we can incorporate it into the url template easily

Check out impersonated builder for reference: https://github.com/googleapis/google-cloud-rust/blob/f8208dd3d008c3a33b06fb3011603868b9f94141/src/auth/src/credentials/impersonated.rs#L180